fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -720,6 +687,32 @@ public Node getNodeVersion(final String label) {
 
     }
 
+    /**
+     * Helps ensure that there are no terminating slashes in the predicate.
+     * A terminating slash means ModeShape has trouble extracting the localName, e.g., for
+     * http://myurl.org/.
+     *
+     * @see <a href="https://jira.duraspace.org/browse/FCREPO-1409"> FCREPO-1409 </a> for details.
+     */
+    private Collection<IllegalArgumentException> checkInvalidPredicates(final UpdateRequest request) {
+        return request.getOperations().stream()
+                .flatMap(x -> {
+                    if (x instanceof UpdateModify) {
+                        final UpdateModify y = (UpdateModify)x;
+                        return Stream.concat(y.getInsertQuads().stream(), y.getDeleteQuads().stream());
+                    } else if (x instanceof UpdateDataInsert || x instanceof UpdateDataDelete) {
+                        return ((UpdateData)x).getQuads().stream();
+                    } else if (x instanceof UpdateDeleteWhere) {
+                        return ((UpdateDeleteWhere)x).getQuads().stream();
+                    } else {
+                        return Stream.empty();
+                    }
```

Comment:
I looked into this (replacing the `instanceof` calls with an `UpdateVisitor`), and the implementation is not particularly simple. If we use the existing jena implementation (`UpdateEngineWorker`), we have to use an interface that is deprecated in 3.x (`GraphStore`) and removed from the `UpdateEngineWorker` interface in 3.x, so when we move to the next jena version, we would probably need to reimplement it. Alternately, we could create a new class that fully implements the `UpdateVisitor` interface, but I don't see how that's any better than what we currently have. And given that this is just a patch until MODE is fixed, I don't think it is worth it.